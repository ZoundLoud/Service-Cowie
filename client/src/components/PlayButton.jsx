import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const PlayButton = (props) => {
  const controller = () => {
    const player = document.getElementById('fredness');
    if (player.paused) {
      player.load();
      player.play();
    } else {
      player.pause();
    }
  };
  return (
    <div className="btnContainer">
      {!props.play ? <FontAwesomeIcon onClick={() => { props.onToggle(); controller(); }} style={{ height: '60px', width: '60px' }} className="playBtn" icon="play-circle" />
        : <FontAwesomeIcon onClick={() => { props.onToggle(); controller(); }} style={{ height: '60px', width: '60px' }} className="playBtn" icon="pause-circle" />
    }
      {
        <audio controls style={{ display: 'none'} } id='fredness' name="media">
          <source id='songsrc' src={props.songUrl.songurl} type="audio/mp3" />
        </audio>
    }
    </div>
  );
};


export default PlayButton;
