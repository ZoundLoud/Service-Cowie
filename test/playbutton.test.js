import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount, shallow, render } from 'enzyme';
import PlayButton from '../client/src/components/PlayButton';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('Play Button Component', () => {
  it('simulates button click for play-button', () => {
    const wrap = shallow(
      <PlayButton />,
    );
    wrap.find('div').simulate('click');
  });
  it('should render Play Button without throwing an error', () => {
    expect(shallow(<PlayButton />).exists()).toBe(true);
  });
  it('Should render correctly according to snapshot', () => {
    const playbutton = shallow(<PlayButton />);
    expect(toJson(playbutton)).toMatchSnapshot();
  });
  it('should render correctly in "debug" mode', () => {
    const debug = shallow(<PlayButton debug />);
  
    expect(toJson(debug)).toMatchSnapshot();
  });
  it('should mount correctly', () => {
    const wrap = mount(<PlayButton />);
    expect(toJson(wrap)).toMatchSnapshot();
  });
  it('should render correctly', () => {
    const wrap = render(<PlayButton />);
    expect(toJson(wrap)).toMatchSnapshot();
  });
});
