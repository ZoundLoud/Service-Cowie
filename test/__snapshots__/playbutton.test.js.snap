// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play Button Component Should render correctly according to snapshot 1`] = `
<div
  className="btnContainer"
>
  <FontAwesomeIcon
    border={false}
    className="playBtn"
    fixedWidth={false}
    flip={null}
    icon="play-circle"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    style={
      Object {
        "height": "60px",
        "width": "60px",
      }
    }
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;

exports[`Play Button Component should mount correctly 1`] = `
<PlayButton>
  <div
    className="btnContainer"
  >
    <FontAwesomeIcon
      border={false}
      className="playBtn"
      fixedWidth={false}
      flip={null}
      icon="play-circle"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      style={
        Object {
          "height": "60px",
          "width": "60px",
        }
      }
      symbol={false}
      title=""
      transform={null}
    />
  </div>
</PlayButton>
`;

exports[`Play Button Component should render correctly 1`] = `
<div
  class="btnContainer"
/>
`;

exports[`Play Button Component should render correctly in "debug" mode 1`] = `
<div
  className="btnContainer"
>
  <FontAwesomeIcon
    border={false}
    className="playBtn"
    fixedWidth={false}
    flip={null}
    icon="play-circle"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    style={
      Object {
        "height": "60px",
        "width": "60px",
      }
    }
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;
